### ISSUE TITLE
Separation of read/write accessors

### ISSUE THREAD

**Issue by abhishiv** [abhishiv/alfama#6]
With reference to #5

**Comment by abhishiv** [comment]
Hey @mindplay-dk, opened a PR for this issue along with a test case. 

I still have to figure out how to implement typescript type for `[Symbol.iterator]: Iterator` and update documentation, but what this is the style react/solid destructuring that you prefer, right?

**Comment by mindplay-dk** [comment]
Yes, but are they accessors? I thought they were just values?

Is it possible to have accessors named `0` and `1`? or were you thinking you would achieve that by implementing `Symbol.iterator`? does iteration happen during an array destructuring assignment? idk.

**Comment by abhishiv** [comment]
Yeah get/set are are just tuple values so you can use them just like in react. I implemented them via Symbol.iterator and seems to work fine. Will merge this over the weekend once I figure how to type it properly.

**Comment by abhishiv** [comment]
Ok, had another go at this PR and looks like we can't use tuples. Based on [this](https://stackoverflow.com/questions/76569775/typescript-how-to-define-an-ordered-iterator-type), typescript doesn't support ordered iterators

> In TypeScript, iterators and generators do not currently support inferring an exact order of yielded values (as of v5.4.5). The type of an iterable is inferred as a union of the types of all possible elements, regardless of the order in which the values are yielded.

So I'm tending on making them get/set accessors. It's a pity since I really liked the tuple value pattern since it allowed me to get rid of $sig naming pattern.

**Comment by abhishiv** [comment]
Ok managed to implement the tuple approach, however haven't released it for now since I think there's a bug. Fixing it in wire PR #10

**Comment by mindplay-dk** [comment]
You can make objects that area also iterable, [like this](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAyglgcwHYEMA2AeAKgPigXigG0AoKKACgEoC8sAaMygN3QFcIAuKLG-PZgHs4AE0YBdKADIoAbyYIIwbtVo8A3EwDOSiqzQduvbkNGaAvppIAzNkgDGwOIKRR7AJwgpgEeMnTYOHrsXDxU3H6omLhyTPYuWsBQisA+7gSUfAIhmuTxSIlQOqkQ6YTBBqHGUKYiarKWcQlJwGxgaNCERClp9EVKaeK5UJ6t7q7y5OQp3D2ljFP9yku9TOREMCAAtgBGgmgAdHBp3oLu4ipZUK3tEBvbe4fHpafn1EzmUChaX0ggFiQSPlCgASfQcDIeLw+SLoCgABioVmBSVI0yUADUQgsoAB6XFQAB6AH5tJiQkx8UTSZJCGCcoDgfsIAc0IIEBQUljKtQkWtFgKqSSgQlmaz2RR6ZUDileeo8QT+FB4fyBWqFdSSEA)
